File: app/src/main/java/com/a_lab/draganddrop/MainFragment.java
package com.a_lab.draganddrop;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;

public class MainFragment extends Fragment {

    private NavController navController;

    @Override
    public void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_main, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        view.findViewById(R.id.btnEasyDrag).setOnClickListener(v -> navController.navigate(R.id.actionMainFragmentToEasyDragFragment));
        view.findViewById(R.id.btnReturnBack).setOnClickListener(v -> navController.navigate(R.id.actionMainFragmentToReturnBackAndFrameHardFragment));
        view.findViewById(R.id.btnTouchConflict).setOnClickListener(v -> navController.navigate(R.id.actionMainFragmentToTouchConflictFragment));
        view.findViewById(R.id.btnClickListener).setOnClickListener(v -> navController.navigate(R.id.actionMainFragmentToClickListenerFragment));
        view.findViewById(R.id.btnIntersectionObjects).setOnClickListener(v -> navController.navigate(R.id.actionMainFragmentToIintersectionObjectsFragment));
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        navController = Navigation.findNavController(requireView());
    }
}

File: app/src/main/java/com/a_lab/draganddrop/MainFragment.kt
package com.a_lab.draganddrop

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController
import com.a_lab.draganddrop.databinding.FragmentMainBinding

class MainFragment : Fragment() {

    private val binding: FragmentMainBinding by lazy {
        DataBindingUtil.inflate(layoutInflater, R.layout.fragment_main, null, false)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ) = binding.root

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val navController = findNavController(requireView())

        binding.easyDragBtn.setOnClickListener {
            navController.navigate(
                R.id.toEasyDragFragment
            )
        }
        binding.returnBackBtn.setOnClickListener {
            navController.navigate(
                R.id.toReturnBackAndFrameHardFragment
            )
        }
        binding.touchConflictBtn.setOnClickListener {
            navController.navigate(
                R.id.toTouchConflictFragment
            )
        }
        binding.clickListenerBtn.setOnClickListener {
            navController.navigate(
                R.id.toClickListenerFragment
            )
        }
        binding.intersectionObjectsBtn.setOnClickListener {
            navController.navigate(
                R.id.toIntersectionObjectsFragment
            )
        }
    }

}